plugins/woocommerce-blocks/assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -205,6 +217,47 @@ const productGallery = {
 				'wc-block-product-gallery-dialog-open'
 			);
 		},
+		onTouchStart: ( event: TouchEvent ) => {
+			const context = getContext();
+			context.touchStartX = event.touches[ 0 ].clientX;
+			context.touchCurrentX = event.touches[ 0 ].clientX;
+			context.isDragging = true;
+		},
+		onTouchMove: ( event: TouchEvent ) => {
+			const context = getContext();
+			if ( ! context.isDragging ) {
+				return;
+			}
+			context.touchCurrentX = event.touches[ 0 ].clientX;
+			event.preventDefault();
+		},
+		onTouchEnd: () => {
+			const context = getContext();
+			if ( ! context.isDragging ) {
+				return;
+			}
+
+			const SNAP_THRESHOLD = 0.3;
+			const delta = context.touchCurrentX - context.touchStartX;
+			const element = getElement()?.ref as HTMLElement;
+			const imageWidth = element?.offsetWidth || 0;
+
+			// Only trigger swipe actions if there was significant movement
+			if ( Math.abs( delta ) > 10 ) {
+				if ( Math.abs( delta ) > imageWidth * SNAP_THRESHOLD ) {
```

Comment:
The first condition comparing with fixed number seems redundant. Unless I'm not getting it?

Reply:
https://github.com/woocommerce/woocommerce/pull/55010/commits/2fc25b939041d48234d532fbdb5fb92809715575